yang/frr-bgp-filter.yang

```diff
@@ -0,0 +1,341 @@
+      range "1..99";
+    }
+    description
+      "Standard list";
+  }
+
+  typedef bgp-list-extended {
+    type uint16 {
+      range "100..500";
+    }
+    description
+      "Extended list";
+  }
+
+  typedef bgp-list-name {
+    type string;
+    description
+      "List name";
+  }
+
+  typedef community-string {
+    type string {
+      pattern "(6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}|6[0-4][0-9]{3}|[1-5][0-9]{4}|[1-9][0-9]{0,3}|0)|((6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}|6[0-4][0-9]{3}|[1-5][0-9]{4}|[1-9][0-9]{0,3}|0):(6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}|6[0-4][0-9]{3}|[1-5][0-9]{4}|[1-9][0-9]{0,3}|0))|(local-AS)|(no-advertise)|(no-export)|(internet)";
+    }
+    description
+      "The BGP community string";
+  }
+
+  typedef large-community-string {
+    type string {
+      pattern "(429496729[0-5]|42949672[0-8][0-9]|4294967[01][0-9]{2}|429496[0-6][0-9]{3}|42949[0-5][0-9]{4}|4294[0-8][0-9]{5}|429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[01][0-9]{8}|[1-3][0-9]{9}|[1-9][0-9]{0,8}|0)|(429496729[0-5]|42949672[0-8][0-9]|4294967[01][0-9]{2}|429496[0-6][0-9]{3}|42949[0-5][0-9]{4}|4294[0-8][0-9]{5}|429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[01][0-9]{8}|[1-3][0-9]{9}|[1-9][0-9]{0,8}|0):(429496729[0-5]|42949672[0-8][0-9]|4294967[01][0-9]{2}|429496[0-6][0-9]{3}|42949[0-5][0-9]{4}|4294[0-8][0-9]{5}|429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[01][0-9]{8}|[1-3][0-9]{9}|[1-9][0-9]{0,8}|0)|(429496729[0-5]|42949672[0-8][0-9]|4294967[01][0-9]{2}|429496[0-6][0-9]{3}|42949[0-5][0-9]{4}|4294[0-8][0-9]{5}|429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[01][0-9]{8}|[1-3][0-9]{9}|[1-9][0-9]{0,8}|0):(429496729[0-5]|42949672[0-8][0-9]|4294967[01][0-9]{2}|429496[0-6][0-9]{3}|42949[0-5][0-9]{4}|4294[0-8][0-9]{5}|429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[01][0-9]{8}|[1-3][0-9]{9}|[1-9][0-9]{0,8}|0):(429496729[0-5]|42949672[0-8][0-9]|4294967[01][0-9]{2}|429496[0-6][0-9]{3}|42949[0-5][0-9]{4}|4294[0-8][0-9]{5}|429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[01][0-9]{8}|[1-3][0-9]{9}|[1-9][0-9]{0,8}|0)";
+    }
+    description
+      "The BGP large-community string";
+  }
+
+  augment "/frr-filter:lib" {
+    list community-list {
+      key "name";
+      description
+        "Community-list instance";
+      leaf name {
+        type string;
+      }
+
+      list entry {
+        key "sequence";
+        description
+          "Community-list entry";
+        leaf sequence {
+          type list-sequence;
+        }
+
+        leaf action {
+          type list-action;
+        }
+
+        leaf type {
+          type enumeration {
+            enum "community-list-standard" {
+              value 0;
+              description
+                "Standard community-list name/identifier";
+            }
+            enum "community-list-extended" {
+              value 1;
+              description
+                "Expanded community-list name/identifier";
+            }
+          }
+          mandatory true;
+          description
+            "Community-list instance name/identifier";
+        }
+
```

Comment:
Why are there four options enumerated in large-community-lists and extcommunity-lists (standard and expanded ID and standard and expanded named) but only the two here combining the name/identifier?